### PR TITLE
Replacing Docker by Podman in the setup

### LINKED ISSUE #81: Add Podman as an alternative for Docker
Docker has limitations in the download/update of Docker Desktop that create issues for a laboratory when many people in parallel use it. 

Podman Is an open-source alternative in which commands work just like Docker from the user's point of view.

### PR DESCRIPTION
**Pull request by michmx**
We are using Podman as the recommendation for the tutorial (but instructions for Docker will stay).

### CODE REVIEW

**Comment by michmx** [comment]
@mambelli I did a test and so far I could install it on MacOS from scratch. Windows I didn't try but the instructions are provided by the Podman team. 
I will go later chapter by chapter to change the commands and add a justification of Podman over Docker.

**Comment by jpivarski** [comment]
@goi42 offered to review these. As soon has he has accepted the org-level invitation, he can be requested as a reviewer.

**Review by jpivarski — APPROVED** [review]
The changes look good to me. Is there a way to see what it looks like when the markdown has been rendered as HTML?

**Comment by michmx** [comment]
> The changes look good to me. Is there a way to see what it looks like when the markdown has been rendered as HTML?

I don't think so... at least not via GitHub pages (@klieret may know a trick that I don't). I did it locally while writing the introduction following the [Carpentries instructions](https://carpentries.github.io/lesson-example/setup.html#setup-for-local-rendering-of-the-lessons-optional).

File: _episodes/01-introduction.md [hunk 1]
```diff
@@ -4,50 +4,106 @@ teaching: 5
...
+Containers are a way to package software that allows you to run an application and its dependencies in a single,
+isolated unit called a container.
+
+Importantly, containers share the host machine's OS system kernel and so don't require an
+OS per application.
+As discrete processes containers take up only as much memory as necessary, making them
+very lightweight and fast to spin up to run:
+
+<figure>
+<img src="../fig/DockerVM.png" alt="missing" style="width:800px"/>
```
Comment (mambelli) [inline _episodes/01-introduction.md]: replace alt definition

File: _episodes/01-introduction.md [hunk 2]
```diff
@@ -4,50 +4,106 @@ teaching: 5
...
+  <figcaption>
+    <i>Container-based architecture vs virtual machines</i>
+  </figcaption>
+</figure>
+
+> ## Containers on Windows and macOS
+>
+> Running containers in systems other than Linux, like macOS and Windows, require a virtual machine on the background to emulate the Linux kernel.
+> Still, the containers are very lightweight and faster to spin up than deploy virtual machines per application.
```
Comment (mambelli) [inline _episodes/01-introduction.md]: maybe "one virtual machine for each application."

File: _episodes/01-introduction.md [hunk 3]
```diff
@@ -4,50 +4,106 @@ teaching: 5
...
-very lightweight and fast to spin up to run.
+Docker is the most popular containerization tool these days, but it's not the only one.
+There are other kids on the block which are in use and gaining popularity, such as Podman.
 
-<img src="../fig/DockerVM.png" alt="DockerVM" style="width:800px">
+# Why Podman?
 
-> ## Singularity
-> Docker is the most popular containerization tool these days, particularly in industry, but it's not the only one. There are other kids on the block including Rocket and Singularity which are in use, but just haven't gained as much large-scale traction.
+<img src="../fig/podman_logo.svg" width="400" alt="Docker logo" />
```
Comment (mambelli) [inline _episodes/01-introduction.md:81]: alt="Podman logo"

File: _episodes/01-introduction.md [hunk 4]
```diff
@@ -4,50 +4,106 @@ teaching: 5
...
+**if you are interested in using Docker instead, just replace `podman` by `docker` in the commands and you should be good to go**.
+
+
+
+> ## Apptainer
+>
+> Apptainer (formerly known as Singularity) is another containerization technology. In particular is used widely in HPC, and it is gaining rapid adoption on High Energy and Nuclear Physics, so you may have need to familiarize yourself with it at some point.
 >
-> Singularity in particular is used widely in HPC, and particularly by CMS, so you may have need to familiarize yourself with it at some point.
+> To learn more about Apptainer, see for example the [HSF Training Singularity Module](https://github.com/hsf-training/hsf-training-singularity-webpage).
```
Comment (mambelli) [inline _episodes/01-introduction.md]: Maybe:
```
[HSF Training Module "Introduction to Apptainer/Singularity"](https://github.com/hsf-training/hsf-training-singularity-webpage) which includes also [more details about the difference between Apptainer and Docker](https://hsf-training.github.io/hsf-training-singularity-webpage/01-introduction/index.html).
```

**Review by mambelli — CHANGES_REQUESTED** [review]
I suggested some changes. Most are optional

File: _episodes/01-introduction.md (hunk 2 above)
Comment (goi42) [reply _episodes/01-introduction.md]: also "than deploy" -> "than to deploy"

File: _episodes/01-introduction.md [hunk 5]
```diff
@@ -4,49 +4,106 @@ teaching: 5
...
+# Why Podman?
 
-> ## Singularity
-> Docker is the most popular containerization tool these days, particularly in industry, but it's not the only one. There are other kids on the block including Rocket and Singularity which are in use, but just haven't gained as much large-scale traction.
+<img src="../fig/podman_logo.svg" width="400" alt="Docker logo" />
+
+Podman is an open-source alternative to Docker with several advantages.
+For example, Podman is able to run containers as a non-root user out of the box, which is a big security advantage over Docker.
+The reason is Podman uses a servless architecture, which means that it doesn't require a daemon running as superuser
+to execute containers as Docker does.
```
Comment (goi42) [inline _episodes/01-introduction.md]: Has "servless architecture" or "daemon" or "superuser" been introduced yet? There should at least be links to explanations of these concepts.

File: setup.md [hunk 6]
```diff
@@ -2,48 +2,84 @@
 title: Setup
 ---
 
-## Docker installations
+## Installations
```
Comment (goi42) [inline setup.md]: "Installation"

File: setup.md [hunk 7]
```diff
@@ -2,48 +2,84 @@
...
+does not require root privileges to use it out of the box. In addition, Docker has licensing restrictions that
+may prevent you from using it in certain sites.
 
-If you are using Linux, then please also follow these [post installation instructions](https://docs.docker.com/engine/install/linux-postinstall/).
+The installation of Podman requires sudo privileges. If you don't have them, check if Podman is already installed on your system
+with
+```bash
+podman version
+```
+And if not available, ask you system administrator to install it for you.
```
Comment (goi42) [inline setup.md]: "If it's not available, ask your system..."

**Comment by goi42** [comment]
Should the title of the lessons be changed? Currently still set to "Introduction to Docker". Similarly, the description is "An opinionated introduction to using Docker as a software development tool. Presented at the February 2020 Analysis Preservation Bootcamp at CERN," which should probably be updated.

File: _config.yml [hunk 8]
```diff
@@ -10,7 +10,7 @@
 carpentry: "hsf"
 
 # Overall title for pages.
-title: "Introduction to Docker"
+title: "Introduction to Podman/Docker"
```
Comment (goi42) [inline _config.yml]: The description (just below the title) is still, "An opinionated introduction to using Docker as a software development tool. Presented at the February 2020 Analysis Preservation Bootcamp at CERN."

File: setup.md [hunk 9]
```diff
@@ -2,48 +2,84 @@
...
 
-~~~bash
-docker pull matthewfeickert/intro-to-docker
-docker pull debian:buster-slim
-docker pull python:2.7-slim
-docker pull python:3.7-slim
-docker pull rootproject/root:6.22.06-conda
-~~~
+Running Podman or Docker on MacOS requires a virtual machine to run the containers.
+In the case of Podman, it provides an installer at https://podman.io/. Download the `.dmg` package for MacOS, extract
```
Comment (goi42) [inline setup.md]: This link is formatted as text (i.e., it is not a clickable URL).

File: setup.md [hunk 10]
```diff
@@ -2,48 +2,84 @@
...
+it and execute Podman Desktop.
 
-## Analysis Code
+The first time that Podman Desktop is executed it will require to
+install Podman and a Podman machine to execute the containers. Follow the instructions at the Setup.
```
Comment (goi42) [inline setup.md]: 'Click "Set up" and follow the instructions.'

File: setup.md [hunk 11]
```diff
@@ -2,48 +2,84 @@
...
+> ## Installing Docker
+>
+> If you still prefer to use Docker (check with the IT department of your institution before using Docker!),
+> follow the official instructions for [Linux](https://docs.docker.com/engine/install/#server), [Mac](https://docs.docker.com/desktop/install/mac-install/), or [Windows](https://docs.docker.com/desktop/install/windows-install/).
+>
+> If you are using Linux, then please also follow these [post installation instructions](https://docs.docker.com/engine/install/linux-postinstall/).
+>
+> Across the tutorial, just replace `podman` by `docker` in the commands and you should be good to go.
+>{: .source}
+{: .callout}
```
Comment (goi42) [inline setup.md]: The way this is formatted makes it look like it's part of the "Install Podman on Windows" section.

File: _episodes/01-introduction.md [hunk 12]
```diff
@@ -4,49 +4,106 @@ teaching: 5
...
+
+In addition, Docker Desktop has [licensing restrictions](https://www.docker.com/pricing/) that may prevent you
+from using it in some institutions. If that is your case, Podman is an excellent alternative.
+
+Podman is a drop-in replacement for Docker, so you can use the same commands and workflows you are used to with Docker.
+Across the tutorial, we will use Podman as the containerization tool, but
+**if you are interested in using Docker instead, just replace `podman` by `docker` in the commands and you should be good to go**.
```
Comment (goi42) [inline _episodes/01-introduction.md:94]: This makes it sound like the following tutorials will show `podman` commands and that students could substitute `podman` -> `docker`. In fact, the tutorials _still_ use `docker` commands, so students will have just been told the tutorials use podman but work with docker when, in fact, it's the other way around.

**Review by goi42 — COMMENTED** [review]
The tutorials beyond the Introduction still use Docker, not Podman--is this intentional?

**Comment by goi42** [comment]
> The tutorials beyond the Introduction still use Docker, not Podman--is this intentional?

Okay, just saw #85 and that these are both part of #81. 👍

File: _episodes/01-introduction.md (hunk 1 above)
Comment (michmx) [reply _episodes/01-introduction.md]: Opps, fixed

File: _episodes/01-introduction.md (hunk 12 above)
Comment (michmx) [reply _episodes/01-introduction.md:94]: Sorry for the confusion. Yes, as you pointed out the commands beyond the introduction are addressed by #85

File: _episodes/01-introduction.md (hunk 5 above)
Comment (michmx) [reply _episodes/01-introduction.md]: That's a very good point, thanks! Addressed at [4f5899fb](https://github.com/hsf-training/hsf-training-docker/pull/82/commits/4f5899fbd8df45349ac12ada6cbb602b20f4db8d)

File: _config.yml (hunk 8 above)
Comment (michmx) [reply _config.yml]: Oh yes! Fixed in the index.md

**Comment by michmx** [comment]
Thanks a lot @mambelli and @goi42 for your careful review and the important fixes suggested! I have addressed all your comments.

**Pull request merged**